src/wazuh_modules/main.c

```diff
@@ -192,15 +183,48 @@ void wm_cleanup()
 
 // Action on signal
 
+void wm_signals_configure()
+{
+    struct sigaction action = { .sa_handler = wm_handler };
+
+    atexit(wm_cleanup);
+    sigaction(SIGTERM, &action, NULL);
+
+    if (flag_foreground) {
+        sigaction(SIGHUP, &action, NULL);
+        sigaction(SIGINT, &action, NULL);
+    }
+
+    action.sa_handler = SIG_IGN;
+    sigaction(SIGPIPE, &action, NULL);
+}
+
 void wm_handler(int signum)
 {
+    static bool wm_sig_processing = false;
+
+    w_mutex_lock(&sig_lock);
+    if (wm_sig_processing) {
+        mdebug2("Signal received: %d, but another one is being processed. Ignoring.", signum);
+        w_mutex_unlock(&sig_lock);
+        return;
+    } else {
+        wm_sig_processing = true;
+        mdebug2("Signal received: %d, handling.", signum);
+    }
+    w_mutex_unlock(&sig_lock);
+
+    // We wait in case the modules haven't completed the initialization
+    sleep(1);
```

Comment:
We know this is not the best approach, but is the best we can do without inquiring about a big refactor regarding the initialization and stopping process of the modules

The stop-and-start interfaces we currently have are not prepared for this kind of behaviour